File: removing_stars_from_a_string.java
class Solution {
    public String removeStars(String s) {
        Stack<Character> st = new Stack<>();

        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if (c == '*') {
                if (!st.isEmpty()) {
                    st.pop();
                }
            } else {
                st.push(c);
            }
        }

        StringBuilder sb = new StringBuilder();
        for (Character c : st) {
            sb.append(c);
        }

        return sb.toString();        
    }
}

File: removing_stars_from_a_string.js
/**
 * @param {string} s
 * @return {string}
 */
var removeStars = function(s) {
    let st = [];

    for (let c of s) {
        if (c === '*') {
            st.pop();
        } else {
            st.push(c);
        }
    }

    return st.join('');    
};
